src/error.rs

```diff
@@ -0,0 +1,30 @@
+//! Errors!
+
+use miette::Diagnostic;
+use thiserror::Error;
+
+/// Gotta love a newtyped Result
+pub type Result<T> = std::result::Result<T, AxoprocessError>;
+
+/// An error from executing a Command
+#[derive(Debug, Error, Diagnostic)]
+pub enum AxoprocessError {
+    /// The command fundamentally failed to execute (usually means it didn't exist)
+    #[error("failed to {summary}")]
+    Exec {
+        /// Summary of what the Command was trying to do
+        summary: String,
+        /// What failed
+        #[source]
+        cause: std::io::Error,
+    },
+    /// The command ran but signaled some kind of error condition
+    /// (assuming the exit code is used for that)
+    #[error("failed to {summary} (status: {status})")]
```

Comment:
```suggestion
    #[error("failed to {summary} (status: {status})")]
```

Waffling on the "failed to" being built into the message. It would arguably just be better/clearer if you had to write that yourself every time.

Reply:
I think it's fine to leave it in.